internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -746,7 +746,8 @@ func shouldUpdateIntegrationTestGitResolver(integrationTestScenario *v1beta2.Int
 
 	if urlVal, ok := params[tekton.TektonResolverGitParamURL]; ok {
 		// urlVal may or may not have git suffix specified :')
-		return urlToGitUrl(urlVal) == urlToGitUrl(annotations[gitops.PipelineAsCodeRepoURLAnnotation])
+		return urlToGitUrl(urlVal) == urlToGitUrl(annotations[gitops.PipelineAsCodeRepoURLAnnotation]) &&
+			params[tekton.TektonResolverGitParamBranch] == annotations[gitops.PipelineAsCodeTargetBranchAnnotation]
```

Comment:
I don't think this would solve the issue from the original [KFLUXBUGS-1340](https://issues.redhat.com/browse/KFLUXBUGS-1340) bug.
The underlying issue was that the `CHAINS-GIT_URL` build pipeline result wasn't reliable enough because it depended on users setting it to be equal to `source_url` PaC variable. 
If the users don't set it that way in their build pipelineRun definition, the above logic fails and causes the behavior in the [KFLUXBUGS-1340](https://issues.redhat.com/browse/KFLUXBUGS-1340) bug.

Reply:
I think that a possible solution would be to update the [getGitResolverUpdateMap](https://github.com/konflux-ci/integration-service/blob/3009a326ac53631063a730ae7b37c5d711b33bc5/internal/controller/snapshot/snapshot_adapter.go#L756C6-L762) function to use the newly introduced PaC `source-repo-url` annotation  [here](https://github.com/konflux-ci/integration-service/blob/3009a326ac53631063a730ae7b37c5d711b33bc5/internal/controller/snapshot/snapshot_adapter.go#L759) instead of the `test.appstudio.openshift.io/source-repo-url` (which is equal to the `CHAINS-GIT_URL` result).

Reply:
thanks for the important correction @dirgim ,   based on your two former comments,
would it be enough to use  PaC annotation `pac.test.appstudio.openshift.io/source-repo-url` without taking in consideration the branch one  `pac.test.appstudio.openshift.io/source-branch`?

Reply:
As far as I'm aware, that should be correct. I don't think that the source branch needs to be considered.